frontend/subrequests/describe.go

```diff
@@ -0,0 +1,63 @@
+package subrequests
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/moby/buildkit/frontend/gateway/client"
+	gwpb "github.com/moby/buildkit/frontend/gateway/pb"
+	"github.com/moby/buildkit/solver/errdefs"
+	"github.com/pkg/errors"
+)
+
+const RequestSubrequestsDescribe = "frontend.subrequests.describe"
+
+var SubrequestsDescribeDefinition = Request{
+	Name:        RequestSubrequestsDescribe,
+	Version:     "1.0.0",
+	Type:        TypeRPC,
+	Description: "List available subrequest types",
+	Metadata: []Named{
+		{
+			Name: "result.json",
+		},
+	},
+}
+
+func Describe(ctx context.Context, c client.Client) ([]Request, error) {
+	gwcaps := c.BuildOpts().Caps
+
+	if err := (&gwcaps).Supports(gwpb.CapFrontendCaps); err != nil {
+		return nil, errdefs.NewUnsupportedSubrequestError(RequestSubrequestsDescribe)
+	}
+
+	res, err := c.Solve(ctx, client.SolveRequest{
+		FrontendOpt: map[string]string{
+			"requestid":     RequestSubrequestsDescribe,
```

Comment:
I find it weird to call this a request ID when it's not really a request ID. Since it's already called `Name` in the `Request` struct, maybe call it `requestname` or `request.name`

Reply:
it is id in the sense that it is a unique identifier for the request that is immutable and may not ever change. As opposed to the name text you would show to the user.

Reply:
Two requests are different by the fact that they were done at different times, and each can have their IDs for audit/debug purposes, that's what I thought of when I saw request ID.

About "name": so the `Name` field is not necessarily always the same as the `requestid` but it is the same in this case ( `RequestSubrequestsDescribe`) ?

Reply:
Looks like `ref` already acts as a request ID in the sense that I would understand it (like https://aws.amazon.com/premiumsupport/knowledge-center/s3-request-id-values/), so even though I find it a bit confusing to call `requestid` something that's more like a type of subrequest, I won't block on this. #namingIsHard